sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -896,8 +895,13 @@ def on_port_update_event(self, port_change_event):
                 self.port_dict[lport]['subport'] = int(port_change_event.port_dict['subport'])
 
             self.force_cmis_reinit(lport, 0)
-        else:
+
+        # PORT_DEL event for the same lport happens 3 times because
+        # we are subscribing to CONFIG_DB, STATE_DB|TRANSCEIVER_INFO, and STATE_DB|PORT_TABLE.
+        # To only handle the first one, check if the lport exists in the port_dict.
+        elif lport in self.port_dict:
             self.update_port_transceiver_status_table_sw_cmis_state(lport, CMIS_STATE_REMOVED)
+            self.port_dict.pop(lport)
```

Comment:
If a transceiver is removed (not doing port breakout), the port would be removed from port_dict. When the transceiver is reattached to the port, there would be no port-related information in port_dict. I think the port should only be removed from port_dict after receiving a delete CONFIG_DB|PORT event and update cmis state after receiving a delete STATE_DB|TRANSCEIVER_INFO event.

Reply:
Thank you for the comment. You're right. I'll update the code as you suggested.

Reply:
@chiourung I fixed the issue with this commit. https://github.com/sonic-net/sonic-platform-daemons/pull/500/commits/954e9cb8f39554c2770c030c796c39feb667c194

Could you take a look?